Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -336,6 +336,7 @@ class BrowserViewController: UIViewController {
         }
         Preferences.NewTabPage.selectedCustomTheme.observe(from: self)
         // Lists need to be compiled before attempting tab restoration
+        ContentBlockerHelper.clearBlockingRules()
```

Comment:
@jumde to fix potential race condition around not waiting for `removeContentRuleList` completion handler